pkg/control/sidecarcontrol/util.go

```diff
@@ -72,9 +73,10 @@ type SidecarSetUpgradeSpec struct {
 }
 
 // PodMatchSidecarSet determines if pod match Selector of sidecar.
-func PodMatchedSidecarSet(pod *corev1.Pod, sidecarSet appsv1alpha1.SidecarSet) (bool, error) {
-	//If matchedNamespace is not empty, sidecarSet will only match the pods in the namespace
-	if sidecarSet.Spec.Namespace != "" && sidecarSet.Spec.Namespace != pod.Namespace {
+func PodMatchedSidecarSet(cl client.Client, pod *corev1.Pod, sidecarSet appsv1alpha1.SidecarSet) (bool, error) {
+	//If pod's namespace in selectedNamespaces, sidecarSet will match the pods in the selectedNamespaces
+	selectedNamespaces := GetNamespacesFromNamespaceSelector(cl, sidecarSet.Spec)
```

Comment:
Is it better to use the following methods ?
```go
selector, err := metav1.LabelSelectorAsSelector(sidecarSet.Spec.NamespaceSelector)
selector.Matches(pod.Namespace)
```

Reply:
`Matches` only support `Matches(label)` such as `Matches(pod.Labels)`, not support Matches(pod.Namespace) 😢

Reply:
> 
Oh, Sorry... 
Maybe, we can just `Get` this namespace of pod instead of `List` ?

Reply:
`selectedNamespaces ` is a  slice so we neeed to list them 🤔